sync/src/types.rs

```diff
@@ -933,16 +921,60 @@ impl<CS: ChainStore> SyncSharedState<CS> {
 
     pub fn mark_as_known_tx(&self, hash: H256) {
         self.inflight_transactions().remove(&hash);
-        self.tx_filter().insert(hash, ());
+        self.tx_filter.lock().insert(hash, ());
     }
 
     pub fn already_known_tx(&self, hash: &H256) -> bool {
-        self.tx_filter().contains_key(hash)
+        self.tx_filter.lock().contains_key(hash)
```

Comment:
Use an inline function instead of a function call can reduce how much time of the lock holding? 

Maybe 1-10ms?

Reply:
I don't really get this comment, care to explain?

Reply:
I just remove the `tx_filter()` method and call `tx_filter.lock()` directly. There are no differences.
So here will not reduce the lock holding time.

I am not sure I get your meaning.

Reply:
I seem to have misread the focus, seeing that the call becomes a function, shortening the time to acquire the lock, but in fact the focus of pr is not here.